packages/aws-cdk-lib/aws-appconfig/lib/environment.ts

```diff
@@ -47,7 +48,35 @@ abstract class EnvironmentBase extends Resource implements IEnvironment, IExtens
   public abstract applicationId: string;
   public abstract environmentId: string;
   public abstract environmentArn: string;
+  public abstract name?: string;
   protected extensible!: ExtensibleBase;
+  protected deploymentQueue: Array<CfnDeployment> = [];
+
+  public addDeployment(configuration: IConfiguration): void {
+    if (this.name === undefined) {
+      throw new Error('Environment name must be known to add a Deployment');
```

Comment:
Just a note.
We can probably change this behavior by using the `environmentId` to build the `CfnDeployment` id.
However, it would require a feature flag and it's out of scope for this PR.